hr_timesheet_overtime/models/hr_timesheet_sheet.py

```diff
@@ -0,0 +1,148 @@
+
+from openerp import api, fields, models, _
+
+
+class HrTimesheetSheet(models.Model):
+    _inherit = "hr_timesheet_sheet.sheet"
+
+    daily_working_hours = fields.Float(
+        "Daily Working Hours",
+        readonly=True,
+        compute="_compute_daily_working_hours",
+        help="Hours to work for the current day",
+    )
+
+    timesheet_overtime = fields.Float(
+        "Timesheet Overtime",
+        readonly=True,
+        store=True,
+        compute="_compute_timesheet_overtime",
+        help="Overtime for this timesheet period",
+    )
+
+    total_overtime = fields.Float(
+        "Overtime Total",
+        readonly=True,
+        store=True,
+        compute="_compute_total_overtime",
+        help="Overtime total since employee's overtime start date",
+    )
+
+    def _compute_daily_working_hours(self):
+        """
+        Computes working hours for the current day according to employee's contracts
+        """
+        for sheet in self:
+            sheet.daily_working_hours = sheet.get_working_hours()
+
+    def _compute_timesheet_overtime(self):
+        """
+        Computes overtime for the timesheet period
+        """
+        for sheet in self:
+            ts_overtime = 0.0
+            total_timesheet_period = sheet.get_total_timesheet_period()
+            for date in total_timesheet_period:
+                daily_wh = sheet.get_working_hours(date=date)
+                daily_overtime = total_timesheet_period[date] - daily_wh
+                ts_overtime += daily_overtime
+
+            sheet.timesheet_overtime = ts_overtime
+
+    def _compute_total_overtime(self):
+        """
+        Computes total overtime since employee's overtime start date
+        """
+        for sheet in self:
+            current_overtime = sheet.employee_id.initial_overtime
+            overtime_start_date = sheet.employee_id.overtime_start_date
+
+            total_timesheet_day = sheet.get_worked_hours_per_day(
+                overtime_start_date
+            )
+            for date in total_timesheet_day:
+                daily_wh = sheet.get_working_hours(date=date)
+                daily_overtime = total_timesheet_day[date] - daily_wh
+                current_overtime += daily_overtime
+
+            sheet.write_hr_employee(current_overtime)
+            sheet.total_overtime = current_overtime
+
+    def get_working_hours(self, date=None):
+        """
+        Get the working hours for a given date according to employee's contracts
+        @param date: string object
+        @return: total of working hours
+        """
+        self.ensure_one()
+        start_dt = None
+        if date:
+            start_dt = fields.Datetime.from_string(date)
+        total = 0.0
+        contracts = self.get_contracts(self.employee_id)
+        for contract in contracts:
+            for calendar in contract.working_hours:
+                total += sum(wh for wh in calendar.get_working_hours_of_date(start_dt=start_dt))
```

Comment:
what happen if date is not defined ?

Reply:
It will use the date returned by `datetime.datetime.now().replace(hour=0, minute=0, second=0)`

From documention in `resource` module :

> if not set, set to end_dt or `today()` if no end_dt at 00.00.00

Reply:
ok !